### PR TITLE
CSSWG note

### LINKED ISSUE #103: Draft and PR a CSSWG note
Using [Bikeshed](https://github.com/speced/bikeshed), draft a document that could look a lot like this [2017 snapshot](https://github.com/w3c/csswg-drafts/blob/main/snapshot-2017-modules.md). It should:

- [ ] enumerate each CSS feature under a heading
- [ ] point to the community group and provide a high level introduction to the purpose
- [ ] have an area for endorsements

Goal is to be a publicly visible, CSSWG official article, on CSS4, CSS5 and beyond. This document should aid in adoption by pacifying concerns about "is this sanctioned or official."

### PR DESCRIPTION
**Pull request by argyleink**
fixes #103 
draft preview: https://css-levels.netlify.app/

todo:
- [x] fix formatting of all the tables
- [x] review text with the group
- [x] update verbiage to use the word "eras" more
- [ ] [PR an edit](https://github.com/w3c/csswg-drafts/blob/2fde733370f069fde148693456c3b6240051c0d3/css-2024/Overview.bs#L651) that links to this note and also adds the note

### CODE REVIEW

File: csswg/levels/index.bs [hunk 1]
```diff
@@ -0,0 +1,841 @@
...
+Markup Shorthands: markdown on
+</pre>
+
+Introduction {#intro}
+=====================
+
+Cascading Style Sheets does not have versions in the traditional sense, 
+instead it has [levels](https://www.w3.org/TR/CSS/#css-levels). 
+Each level of CSS builds on the previous module, refining definitions and adding features. 
+In Oct 27, 2023, the
```
Comment (Que-tin) [inline csswg/levels/index.bs:23]: On

File: csswg/levels/index.bs [hunk 2]
```diff
@@ -0,0 +1,841 @@
...
+Shortname: css-levels
+Level: 1
+Previous Version: https://www.w3.org/TR/CSS/#css-level
+Status: w3c/ED
+Group: CSS-Next
+Repository: CSS-Next/css-next
+ED: https://drafts.csswg.org/CSS/levels/
+Editor: Una Kravets, Google, https://una.im, w3cid 115525
+Editor: Adam Argyle, Google, https://nerdy.dev, w3cid 112669
+Editor: Brecht De Ruyte, IO Digital, https://utilitybend.com/
```
Comment (Que-tin) [inline csswg/levels/index.bs:12]: Editor: Quentin Albert, beyond-the-cascade.com

File: csswg/levels/index.bs [hunk 3]
```diff
@@ -0,0 +1,841 @@
...
+Abstract: This document outlines where CSS levels have been and where they are going. It categorizes CSS properties into distinct levels, namely CSS4, and CSS5, to better organize and facilitate understanding of the evolving CSS landscape. The primary audience is teachers, learners, and implementers of CSS.
```
Comment (Que-tin) [inline csswg/levels/index.bs:13]: Do we need _"The primary audience is teachers, learners, and implementers of CSS"_? I would leave it out.

File: csswg/levels/index.bs [hunk 4]
```diff
@@ -0,0 +1,841 @@
...
+The term "CSS3" has been widely used to encompass various additions and enhancements 
+to Cascading Style Sheets (CSS) since around 2010. 
+Originally, the CSSWG defined CSS3 as "everything after CSS2". 
+As CSS continues to evolve with new features and specifications, 
+the blanket term "CSS3" has become insufficient and misleading. 
+[This RFC](https://github.com/CSS-Next/css-next/discussions/92) from the CSS Next Community Group 
+proposes the categorization of CSS properties into distinct groups, namely CSS3, CSS4, and CSS5, 
+to better organize and facilitate understanding of the evolving CSS landscape. 
+
+We've found that calling CSS3, CSS4, and CSS5 as "eras," are helping people
```
Comment (Que-tin) [inline csswg/levels/index.bs:36]: _We've_ sounds weird here as everything else is written in 3rd person

File: csswg/levels/index.bs [hunk 5]
```diff
@@ -0,0 +1,841 @@
...
+These eras aim to improve adoption and ease of teaching, while not having a 
+direct impact towards the CSS Working Group (CSSWG) operations or the 
+[Baseline initiative](https://github.com/web-platform-dx/web-features/blob/main/docs/baseline.md).
+
+Categorization {#categorization}
+================================
+
+Categorization is based on a variety of factors 
+with the primary factor being the general timeline that an API was initially specified 
+in the CSS Working Group.
```
Comment (Que-tin) [inline csswg/levels/index.bs:47]: Do we need to define _initially specified _? For us it was everything post Editors Draft

File: csswg/levels/index.bs [hunk 6]
```diff
@@ -0,0 +1,841 @@
...
+direct impact towards the CSS Working Group (CSSWG) operations or the 
+[Baseline initiative](https://github.com/web-platform-dx/web-features/blob/main/docs/baseline.md).
+
+Categorization {#categorization}
+================================
+
+Categorization is based on a variety of factors 
+with the primary factor being the general timeline that an API was initially specified 
+in the CSS Working Group. 
+Additionally, implementer interest and adoption were moderately taken into account
```
Comment (Que-tin) [inline csswg/levels/index.bs:48]: We didn't take user _ adoption_ into account rather than vendor adoption, right? At least I cannot think about a single case.
Should we clarify that?

File: csswg/levels/index.bs [hunk 7]
```diff
@@ -0,0 +1,841 @@
...
+
+Categorization is based on a variety of factors 
+with the primary factor being the general timeline that an API was initially specified 
+in the CSS Working Group. 
+Additionally, implementer interest and adoption were moderately taken into account 
+in the categorization process.
+
+* **CSS3:** This category was established before this community group began articulating levels 
+    and was defined by the CSSWG, and was designed to be open ended. 
+    It serves as a historical reference point for CSS advancements up to a certain point.
```
Comment (Que-tin) [inline csswg/levels/index.bs:53]: The certain point in our levels is 2012, would probably make sense to list it, as all others levels have their yeas listed accordingly

File: csswg/levels/index.bs [hunk 8]
```diff
@@ -0,0 +1,841 @@
...
+* **CSS4:** Properties and features introduced in a spec **around 2013 - 2018**.
+* **CSS5:** Properties and features introduced in a spec **around 2019 - 2024**.
+* **Future/Next**: This category is a placeholder for properties and features that 
+    are currently in development or proposed for future versions of CSS beyond CSS5.
```
Comment (Que-tin) [inline csswg/levels/index.bs:57]: versions -> levels

File: csswg/levels/index.bs [hunk 9]
```diff
@@ -0,0 +1,841 @@
...
+
+List of CSS4 and CSS5 features {#feature-list}
+==============================================
+
+This list should be considered a work in progress.
+
+CSS4 {#css-4}
```
Comment (Que-tin) [inline csswg/levels/index.bs:64]: This should be 3.1

File: csswg/levels/index.bs [hunk 10]
```diff
@@ -0,0 +1,841 @@
...
+| [text-emphasis-position](https://developer.mozilla.org/en-US/docs/Web/CSS/text-emphasis-position)                         |       |
+| [text-emphasis-style](https://developer.mozilla.org/en-US/docs/Web/CSS/text-emphasis-style)                               |       |
+| [text-orientation](https://developer.mozilla.org/en-US/docs/Web/CSS/text-orientation)                                     |       |
+| [text-underline-offset](https://developer.mozilla.org/en-US/docs/Web/CSS/text-underline-offset)                           |       |
+| [text-underline-position](https://developer.mozilla.org/en-US/docs/Web/CSS/text-underline-position)                       |       |
+| ornaments()                                                                                                               |       |
+
+</details>
+
+CSS5 {#css-5}
```
Comment (Que-tin) [inline csswg/levels/index.bs:473]: This should be 3.2

File: csswg/levels/index.bs [hunk 11]
```diff
@@ -0,0 +1,841 @@
...
+| [::view-transition-new](https://developer.mozilla.org/en-US/docs/Web/CSS/::view-transition-new)               |       |
+| [::view-transition-old](https://developer.mozilla.org/en-US/docs/Web/CSS/::view-transition-old)               |       |
+| @view-transition                                                                                              |       |
+| view-transition-class                                                                                         |       |
+| [view-transition-name](https://developer.mozilla.org/en-US/docs/Web/CSS/view-transition-name)                 |       |
+| view-transition-type                                                                                          |       |
+
+</details>
+
+Future (Uncategorized, out of scope for CSS4/5) {#css-uncategorized}
```
Comment (Que-tin) [inline csswg/levels/index.bs:714]: This should be 3.3

File: csswg/levels/index.bs [hunk 12]
```diff
@@ -0,0 +1,841 @@
...
+
+* **CSS3:** This category was established before this community group began articulating levels 
+    and was defined by the CSSWG, and was designed to be open ended. 
+    It serves as a historical reference point for CSS advancements up to a certain point.
+* **CSS4:** Properties and features introduced in a spec **around 2013 - 2018**.
+* **CSS5:** Properties and features introduced in a spec **around 2019 - 2024**.
+* **Future/Next**: This category is a placeholder for properties and features that 
+    are currently in development or proposed for future versions of CSS beyond CSS5.
+
+List of CSS4 and CSS5 features {#feature-list}
```
Comment (Que-tin) [inline csswg/levels/index.bs:59]: Lists of Level-Features

I mean we also have a list of Future features.

File: csswg/levels/index.bs (hunk 7 above)
Comment (fantasai) [reply csswg/levels/index.bs:56]: Agree, it's not "defined by the CSSWG" anymore, it's defined in this document.

File: csswg/levels/index.bs [hunk 13]
```diff
@@ -0,0 +1,747 @@
+<pre class='metadata'>
+Title: CSS Eras
+Shortname: css-eras
+Level: 1
+Previous Version: https://www.w3.org/TR/CSS/#css-level
+Status: w3c/ED
```
[the remark this answers is not in the document]
Comment (argyleink) [reply csswg/levels/index.bs:6]:
```suggestion
Status: CG-DRAFT
```

nice ty! like this?

File: csswg/levels/index.bs [hunk 14]
```diff
@@ -0,0 +1,747 @@
+<pre class='metadata'>
+Title: CSS Eras
+Shortname: css-eras
```
[the remark this answers is not in the document]
Comment (argyleink) [reply csswg/levels/index.bs:3]: agree it's worth greater discussion. 

we changed all the references to eras so we don't muddy the meaning or original intent of levels. but it's also a decently small aspect and could yeah, live as just a term in prose.

good place for that issue here in this repo? or better for the wider CSSWG repo?

File: csswg/levels/index.bs [hunk 15]
```diff
@@ -0,0 +1,747 @@
...
+Cascading Style Sheets does not have versions in the traditional sense, 
+instead it had [levels](https://www.w3.org/TR/CSS/#css-levels). 
+Each level of CSS builds on the previous module, refining definitions and adding features. 
+On Oct 27, 2023, the 
+[CSS Working Group resolved to "support the definition of CSS4"](https://github.com/w3c/csswg-drafts/issues/4770#issuecomment-1802342332) 
+and delegated the categorization of CSS properties to the [CSS Next Community Group](https://github.com/CSS-Next/css-next).
+
+The term "CSS3" has been widely used to encompass various additions and enhancements 
+to Cascading Style Sheets (CSS) since around 2010. 
+Originally, the CSSWG defined CSS3 as "everything after CSS2".
```
[the remark this answers is not in the document]
Comment (argyleink) [reply csswg/levels/index.bs:33]:
```suggestion
The CSSWG defined CSS3 as "everything after CSS2". 
```

No worries, easy to fix. Simply removing the word from the sentence is all that's needed yeah?

File: csswg/levels/index.bs (hunk 7 above)
Comment (argyleink) [reply csswg/levels/index.bs:56]:
```suggestion
    It serves as a historical reference point for CSS advancements up to 2012.
```

We like this better? Or should the whole line be removed?

**Review by argyleink — COMMENTED** [review]
thanks for the reviews all, some suggestions and edits have been made 🤘🏻

File: csswg/levels/index.bs (hunk 14 above)
[the remark this answers is not in the document]
Comment (fantasai) [reply csswg/levels/index.bs:3]: Probably in this repo. We can flag it for the CSSWG if that seems appropriate.

File: csswg/levels/index.bs (hunk 7 above)
Comment (fantasai) [reply csswg/levels/index.bs:56]: Just remove it. And inline an actual definition, just like the other levels.

File: csswg/levels/index.bs [hunk 16]
```diff
@@ -0,0 +1,747 @@
...
+
+Eras {#eras}
+================================
+
+Categorization is based on a variety of factors 
+with the primary factor being the general timeline that an API was specified 
+in the CSS Working Group. 
+Additionally, implementer interest and implementer adoption were moderately taken into account.
+
+* **CSS3** <br>
```
Comment (fantasai) [inline csswg/levels/index.bs:53]: Use a <dl>?

Also some amount of thematic description might be helpful. I think that was part of how you developed the categories, right?

File: csswg/levels/index.bs (hunk 7 above)
Comment (argyleink) [reply csswg/levels/index.bs:56]:
```suggestion
```

File: csswg/levels/index.bs (hunk 16 above)
Comment (argyleink) [reply csswg/levels/index.bs]: <dl>, great idea, found the bikeshed syntax, updated!

also added prose to categorization, to help clarify some of the process the CG went through.

**Comment by brechtDR** [comment]
Hi all, just a small bit of feedback, just to confirm:

> The CSSWG defined CSS3 as "everything after CSS2"

To make it perfectly on point, should that be CSS2.1 ? 
[Based on this link](https://www.w3.org/TR/CSS2/about.html#:~:text=CSS%202.1%20is%20derived%20from,in%20a%20future%20CSS3%20specification.)

> It’s been helpful for people to call CSS3, CSS4, and CSS5 as "eras,"

Just wondering if we should still mention the word era, I like it, but I know some had the feedback of an era being more a historical term. as an era is "a period of time beginning with some special date or event". 
But this could become a special date of course

**Comment by subuhunter** [comment]
Agree with @brechtDR 's concern on era. But other than that this looks perfect IMO.

**Review by brechtDR — APPROVED** [review]

File: csswg/levels/index.bs [hunk 17]
```diff
@@ -45,10 +45,18 @@ direct impact on the CSS Working Group (CSSWG) or the
 Categorization {#categorization}
 ================================
 
-Categorization is based on a variety of factors 
-with the primary factor being the general timeline that an API was specified 
-in the CSS Working Group. 
-Additionally, implementer interest and implementer adoption were moderately taken into account.
+Feature categorization is based on a variety of factors like when it was proposed, 
+when it was implemented, and when it was stabilized.
```
Comment (Que-tin) [inline csswg/levels/index.bs:49]: and when it ~~was~~ stabilized.

Maybe it hasn't stabilized yet.

Comment (Que-tin) [reply csswg/levels/index.bs:49]: I also feel like there is a comma missing. I would rephrase it and fix it like that:

Feature categorization is based on a variety of factors, such as when it was proposed, 
when it was implemented, and when it stabilized.

File: csswg/levels/index.bs [hunk 18]
```diff
@@ -45,10 +45,18 @@ direct impact on the CSS Working Group (CSSWG) or the
...
+The primary factor being the general timeline that a feature stabalized 
+in the CSS Working Group. The community group inspected each CSS feature's spec,
```
Comment (Que-tin) [inline csswg/levels/index.bs:51]: No shortforms spec -> specification (just for clearance)

File: csswg/levels/index.bs [hunk 19]
```diff
@@ -45,10 +45,18 @@ direct impact on the CSS Working Group (CSSWG) or the
...
-Categorization is based on a variety of factors 
-with the primary factor being the general timeline that an API was specified 
-in the CSS Working Group. 
-Additionally, implementer interest and implementer adoption were moderately taken into account.
+Feature categorization is based on a variety of factors like when it was proposed, 
+when it was implemented, and when it was stabilized.
+The primary factor being the general timeline that a feature stabalized 
+in the CSS Working Group. The community group inspected each CSS feature's spec, 
+looked at the date of inception and date of stability, and used this information 
+as a corner stone attribute for categorization.
```
Comment (Que-tin) [inline csswg/levels/index.bs:53]: ~~corner stone~~ -> cornerstone

File: csswg/levels/index.bs [hunk 20]
```diff
@@ -45,10 +45,18 @@ direct impact on the CSS Working Group (CSSWG) or the
...
+
+The primary focus is on CSS; not CSS as it is implemented in browsers.
```
Comment (Que-tin) [inline csswg/levels/index.bs:55]: The primary focus is on CSS **as a language**; not it's adoption inside of browsers.

(probably need to be rephrased, but reads more clear imo)

File: csswg/levels/index.bs [hunk 21]
```diff
@@ -45,10 +45,18 @@ direct impact on the CSS Working Group (CSSWG) or the
...
-Additionally, implementer interest and implementer adoption were moderately taken into account.
+Feature categorization is based on a variety of factors like when it was proposed, 
+when it was implemented, and when it was stabilized.
+The primary factor being the general timeline that a feature stabalized 
+in the CSS Working Group. The community group inspected each CSS feature's spec, 
+looked at the date of inception and date of stability, and used this information 
+as a corner stone attribute for categorization.
+
+The primary focus is on CSS; not CSS as it is implemented in browsers. 
+When applicable though, implementer interest and browser adoption were taken into account
```
Comment (Que-tin) [inline csswg/levels/index.bs:56]: ~~were~~ -> are

This list is open for updates and therefore not finished

File: csswg/levels/index.bs [hunk 22]
```diff
@@ -45,10 +45,18 @@ direct impact on the CSS Working Group (CSSWG) or the
...
+when it was implemented, and when it was stabilized.
+The primary factor being the general timeline that a feature stabalized 
+in the CSS Working Group. The community group inspected each CSS feature's spec, 
+looked at the date of inception and date of stability, and used this information 
+as a corner stone attribute for categorization.
+
+The primary focus is on CSS; not CSS as it is implemented in browsers. 
+When applicable though, implementer interest and browser adoption were taken into account 
+and used to resolve difficult to place features. For a robust categorization of 
+browser support, see the
```
Comment (Que-tin) [inline csswg/levels/index.bs:58]: @argyleink shouldn't it be

For a solid categorization **according to** browser support

I would propbably rephrase it like this:

For a CSS-Feature categorization **according to** browser support

File: csswg/levels/index.bs [hunk 23]
```diff
@@ -0,0 +1,754 @@
...
+to Cascading Style Sheets (CSS) since around 2010. 
+The CSSWG defined CSS3 as "everything after CSS2". 
+As CSS continues to evolve with new features and specifications, 
+the blanket term "CSS3" has become insufficient and misleading. 
+[This RFC](https://github.com/CSS-Next/css-next/discussions/92) from the CSS Next Community Group 
+proposes the categorization of CSS properties into distinct levels; namely CSS3, CSS4, and CSS5. 
+
+It's been helpful for people to call CSS3, CSS4, and CSS5 as "eras," as this helps  
+comparmentalize the intent of the categorizations.  
+These eras aim to improve adoption and ease of teaching, while not having a
```
Comment (Que-tin) [inline csswg/levels/index.bs:41]: As @brechtDR noted, I'm not quite sure if we should keep the eras after the latest feedback from @fantasai. Could lead to confusion introducing a new term.

File: csswg/levels/index.bs (hunk 17 above)
Comment (argyleink) [reply csswg/levels/index.bs:49]: will add the suggesting in a follow up commit 👍🏻

File: csswg/levels/index.bs (hunk 18 above)
Comment (argyleink) [reply csswg/levels/index.bs:51]: will add the suggesting in a follow up commit 👍🏻

File: csswg/levels/index.bs (hunk 19 above)
Comment (argyleink) [reply csswg/levels/index.bs:53]: will add the suggesting in a follow up commit 👍🏻

File: csswg/levels/index.bs (hunk 20 above)
Comment (argyleink) [reply csswg/levels/index.bs:55]: will add the suggesting in a follow up commit 👍🏻

File: csswg/levels/index.bs (hunk 23 above)
Comment (argyleink) [reply csswg/levels/index.bs:41]: she said that era's is fine to mention, but warned of trying to change entirely away from levels. we can mention it 👍🏻

**Review by Que-tin — APPROVED** [review]

**Review by subuhunter — APPROVED** [review]

**Review by rol4nd909 — APPROVED** [review]
Looks good! Go for it.

**Review by Amit-Sheen — APPROVED** [review]

**Review by rol4nd909 — APPROVED** [review]

File: csswg/levels/index.bs [hunk 24]
```diff
@@ -0,0 +1,762 @@
...
+These eras aim to improve adoption and ease of teaching, while not having a 
+direct impact on the CSS Working Group (CSSWG) or the 
+[Baseline initiative](https://github.com/web-platform-dx/web-features/blob/main/docs/baseline.md).
+
+Categorization {#categorization}
+================================
+
+Feature categorization is based on a variety of factors, such as when it was proposed,
+when it was implemented, and when it stabilized. 
+The primary factor being the general timeline that a feature stabalized
```
Comment (stof) [inline csswg/levels/index.bs:59]:
```suggestion
The primary factor being the general timeline that a feature stabilized 
```

**Comment by svgeesus** [comment]
@argyleink @una This PR looks to have approvals, is it waiting for something else or should it be merged?

**Comment by argyleink** [comment]
> @argyleink @una This PR looks to have approvals, is it waiting for something else or should it be merged?

we never fully wrapped it up. it got tons of time, attention and work, but couldnt quite get carried across the finish line. we can close for now, maybe another will try again.
